samples/azure/blob-storage-databus-cleanup-function/ABSDataBus_4/DataBusBlobCleanupFunctions/DataBusCleanupOrchestrator.cs

```diff
@@ -0,0 +1,36 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.DurableTask;
+using Microsoft.Extensions.Logging;
+
+public static class DataBusCleanupOrchestrator
+{
+    #region DataBusCleanupOrchestratorFunction
+
+    [FunctionName(nameof(DataBusCleanupOrchestrator))]
+    public static async Task RunOrchestrator([OrchestrationTrigger] IDurableOrchestrationContext context, ILogger log)
+    {
+        var blobData = context.GetInput<DataBusBlobData>();
+
+        log.LogInformation($"Orchestrating deletion for blob at {blobData.Path} with ValidUntilUtc of {blobData.ValidUntilUtc}");
+
+        var validUntilUtc = DataBusBlobTimeoutCalculator.ToUtcDateTime(blobData.ValidUntilUtc);
+
+        DateTime timeoutUntil;
+
+        //Timeouts currently have a 7 day limit, use 6 day loops until the wait is less than 6 days
+        do
+        {
+            timeoutUntil = validUntilUtc > context.CurrentUtcDateTime.AddDays(2) ? context.CurrentUtcDateTime.AddDays(2) : validUntilUtc;
+            log.LogInformation($"Waiting until {timeoutUntil}/{validUntilUtc} for blob at {blobData.Path}. Currently {context.CurrentUtcDateTime}.");
+            await context.CreateTimer(DataBusBlobTimeoutCalculator.ToUtcDateTime(blobData.ValidUntilUtc), CancellationToken.None);
```

Comment:
can't you use here directly `validUntilUtc` again?

Reply:
What do you mean?

Reply:
we have `DataBusBlobTimeoutCalculator.ToUtcDateTime(blobData.ValidUntilUtc)` twice. Once here and once above. So we already parsed the time into the local `validUntilUtc` right?

Reply:
Ah, yeah, that was the original logic and I'm not sure if it was intentional or just an oversight. I'll verify in the next round of changes if we can reduce it to a single variable.